limits/produce_limits_allCombined.py

```diff
@@ -0,0 +1,337 @@
+
+# import pickle
+# with open('results.pck', 'r') as ff:
+#     limits2D = pickle.load(ff)
+
+import pickle
+with open('results.pck', 'w') as ff:
+    pickle.dump(limits2D, ff)
+
+
+masses_obs       = []
+masses_central   = []
+masses_one_sigma = []
+masses_two_sigma = []
+
+minus_two = []
+minus_one = []
+central   = []
+plus_one  = []
+plus_two  = []
+obs       = []
+
+# go through the different mass points first left to right to catch the lower exclusion bound
+# then right to left to catch the upper exclusion bound
+for mass in sorted(limits2D.keys()):
+
+    if len(limits2D[mass]['obs'])>0: 
+        obs.append( min(limits2D[mass]['obs']) )
+        masses_obs.append(mass)
+    
+    if len(limits2D[mass]['exp_central'])>0: 
+        central.append( min(limits2D[mass]['exp_central']) )
+        masses_central.append(mass)
+
+    if len(limits2D[mass]['exp_minus_one'])>0 and len(limits2D[mass]['exp_plus_one' ])>0: 
+        minus_one.append( min(limits2D[mass]['exp_minus_one']) )
+        plus_one .append( min(limits2D[mass]['exp_plus_one' ]) )
+        masses_one_sigma.append(mass)
+
+    if len(limits2D[mass]['exp_minus_two'])>0 and len(limits2D[mass]['exp_plus_two' ])>0: 
+        minus_two.append( min(limits2D[mass]['exp_minus_two']) )
+        plus_two .append( min(limits2D[mass]['exp_plus_two' ]) )
+        masses_two_sigma.append(mass)
+    
+for mass in sorted(limits2D.keys(), reverse=True):
+
+    if len(limits2D[mass]['obs'])>1: 
+        obs.append( max(limits2D[mass]['obs']) )
+        masses_obs.append(mass)
+    
+    if len(limits2D[mass]['exp_central'])>1: 
+        central.append( max(limits2D[mass]['exp_central'  ]) )
+        masses_central.append(mass)
+
+    if len(limits2D[mass]['exp_minus_one'])>1 and len(limits2D[mass]['exp_plus_one' ])>1: 
+        minus_one       .append( max(limits2D[mass]['exp_minus_one']) )
+        plus_one        .append( max(limits2D[mass]['exp_plus_one' ]) )
+        masses_one_sigma.append(mass)
+
+    if len(limits2D[mass]['exp_minus_two'])>1 and len(limits2D[mass]['exp_plus_two' ])>1: 
+        minus_two       .append( max(limits2D[mass]['exp_minus_two']) )
+        plus_two        .append( max(limits2D[mass]['exp_plus_two' ]) )
+        masses_two_sigma.append(mass)
+
+# plot the 2D limits
+plt.clf()
+
+plt.fill_between(masses_two_sigma, minus_two, plus_two, color='gold'       , label=r'$\pm 2 \sigma$')
+plt.fill_between(masses_one_sigma, minus_one, plus_one, color='forestgreen', label=r'$\pm 1 \sigma$')
+plt.plot        (masses_central  , central            , color='red'        , label='central expected', linewidth=2)
+plt.plot        (masses_obs      , obs                , color='black'      , label='observed', linewidth=2)
+
+plt.ylabel(flavour)
+plt.xlabel('mass (GeV)')
+plt.legend()
+plt.ticklabel_format(axis='y', style='sci', scilimits=(0,0))
+#plt.tight_layout()
+plt.yscale('log')
+plt.xscale('linear')
+plt.grid(True)
+plt.savefig('2d_hnl_limit.pdf')
+
```

Comment:
general comments:
- instead of having clones of basically the same file for the single channel, per year combo and multiyear combo, it would be great if we could make this configurable via arg parser
- what happens if a signal mass point is present in one year but not in another (this happens as not all the samples were produced for all years)
- another nice-have: decouple the plotting part and play a bit with splines (maybe this is one instance where ROOT does better)